### ISSUE TITLE
bug: Read MT3DMS and MT3DUSGS Budgets from List File fails width ADV-Package activated

### ISSUE THREAD

**Issue by rabbl** [modflowpy/flopy#1895]
**Describe the bug**
Reading the MassBudgets from a MT3D-List file fails.

We are seeing two bugs here. 

1. Reading out "CALLS TO GCG PACKAGE" (see attached file line 249) fails, because the expected character span (line[51:58]) does not match with the existing here. 

The sourcecode tries to parse, but returns more characters than expected.
```
line = '     1 CALLS TO GCG PACKAGE FOR TRANSPORT TIME STEP   1 IN FLOW TIME STEP   1 STRESS PERIOD   1'
print(line[51:58])

>>> '   1 IN'
```

2. With activated ADV-Package, three additional lines regarding Particle Tracking are created for each Component and Step. Instead of parsing. 

The sourcecode expects to parse:

`                     CUMMULATIVE MASS BUDGETS AT END OF TRANSPORT STEP   21, TIME STEP    1, STRESS PERIOD    1` 

but it gets

`TOTAL PARTICLES USED IN THE CURRENT STEP =      4264`

**To Reproduce**

Attached goes the mt.list as zip-file.

**Additional context**

To solve the first bug I propose to use regex instead of static character spans.
This could be something like that, reading a number which in in between known strings.

```python
   def _extract_number_between_strings(input_string, start_string, end_string):
        pattern = rf"{re.escape(start_string)}\s*(\d+)\s*{re.escape(end_string)}"
        match = re.search(pattern, input_string)
        if match:
            return int(match.group(1))
        else:
            raise Exception(
                f"Error extracting number between {start_string} and {end_string} in {input_string}"
            )
```

To solve the second bug, I propose to add a simple conditional to skip 4 more lines if the string contains part of "TOTAL PARTICLES USED IN THE CURRENT STEP".

We will provide a PR.

[mt.list.zip](https://github.com/modflowpy/flopy/files/12240066/mt.list.zip)

**Comment by rabbl** [comment]
I have created #1899 providing a fix with tests.

**Comment by wpbonelli** [comment]
presumed closed by #1899, please reopen if not.
